src/splitobs.jl

```diff
@@ -43,54 +38,32 @@ function _splitobs(n::Int, at::NTuple{N,<:AbstractFloat}) where N
 end
 
 """
-    splitobs(data; at) -> Tuple
+    splitobs(data; at, shuffle=false) -> Tuple
 
 Split the `data` into multiple subsets proportional to the
-value(s) of `at`.
+value(s) of `at`. 
+
+If `shuffle=true`, randomly permute the observations before splitting.
 
-Note that this function will perform the splits statically and
-thus not perform any randomization. The function creates a
-`NTuple` of data subsets in which the first N-1 elements/subsets
-contain the fraction of observations of `data` that is specified
-by `at`.
+Supports any datatype implementing the [`numobs`](@ref) and
+[`getobs`](@ref) interfaces.
 
-For example, if `at` is a `Float64` then the return-value will be
-a tuple with two elements (i.e. subsets), in which the first
-element contains the fracion of observations specified by `at`
-and the second element contains the rest. In the following code
-the first subset `train` will contain the first 70% of the
-observations and the second subset `test` the rest.
+# Examples
 
 ```julia
+# A 70%-30% split
 train, test = splitobs(X, at=0.7)
-```
-
-If `at` is a tuple of `Float64` then additional subsets will be
-created. In this example `train` will have the first 50% of the
-observations, `val` will have next 30%, and `test` the last 20%
 
-```julia
+# A 50%-30%-20% split
 train, val, test = splitobs(X, at=(0.5, 0.3))
-```
-
-It is also possible to call `splitobs` with multiple data
-arguments as tuple, which all must have the same number of total
-observations. This is useful for labeled data.
 
-```julia
-train, test = splitobs((X, y), at=0.7)
-(x_train,y_train), (x_test,y_test) = splitobs((X, y), at=0.7)
-```
-
-If the observations should be randomly assigned to a subset,
-then you can combine the function with `shuffleobs`
-
-```julia
-# This time observations are randomly assigned.
-train, test = splitobs(shuffleobs((X, y)), at=0.7)
+# A 70%-30% split with multiple arrays and shuffling
+train, test = splitobs((X, y), at=0.7, shuffle=true)
+Xtrain, Ytrain = train
 ```
 """
-function splitobs(data; at)
+function splitobs(data; at, shuffle=false)
+    shuffle && return splitobs(shuffleobs(data); at, shuffle=false)
```

Comment:
It seems rather unfortunate to lose type stability over a disabled by default kwarg. Does 
```suggestion
    if shuffle
      data = shuffleobs(data)
    end
```
Change anything?

Reply:
agree `shuffle=true` is an important convenience kwarg!
the `if shuffle (...) end` syntax is much clearer to me anyway.
also, what is the reasoning for not writing `shuffle::Bool=false`? (just wondering--does this not help the compiler?)

Reply:
I don't know if it does or not, but it might be worth adding as a user-facing thing so that we fail fast on an invalid kwarg value.

Reply:
added the proposed change but didn't help with instability. I think we have to live with it for the time being